### PR TITLE
Add xsimd::polar()

### PR DESCRIPTION
**Pull request by jatinchowdhury18**
Intended to be the equivalent of `std::polar()`

### CODE REVIEW

File: include/xsimd/arch/generic/xsimd_generic_math.hpp [hunk 1]
```diff
@@ -962,6 +962,21 @@ namespace xsimd
...
+        // polar
+        template <class A, class T>
+        inline batch<std::complex<T>, A> polar(const batch<T, A>& r, const batch<T, A>& theta, requires_arch<generic>) noexcept
+        {
+            auto sincosTheta = sincos(theta);
+            return { r * sincosTheta.second, r * sincosTheta.first };
+        }
+
+        template <class A, class T>
+        inline batch<std::complex<T>, A> polar(const batch<T, A>& theta, requires_arch<generic>) noexcept
```
Comment (serge-sans-paille) [inline include/xsimd/arch/generic/xsimd_generic_math.hpp:974]: shouldn't this be `r` instead of `theta`?

Comment (serge-sans-paille) [reply include/xsimd/arch/generic/xsimd_generic_math.hpp:974]: And then the formula is different, of course. But that would match the C++ api, as https://en.cppreference.com/w/cpp/numeric/complex/polar

File: include/xsimd/types/xsimd_api.hpp [hunk 2]
```diff
@@ -1334,6 +1334,33 @@ namespace xsimd
...
+    {
+        return kernel::polar<A>(r, theta, A {});
+    }
+
+    /**
+     * @ingroup batch_math
+     *
+     * Returns a complex batch with phase angle \c theta.
+     * @param theta The phase angle of the desired complex result.
+     * @return exp(i * \c theta).
```
Comment (serge-sans-paille) [inline include/xsimd/types/xsimd_api.hpp:1356]: This is not the standard API, we try to stick to `std::polar`, even if that misses an optimization

**Review by serge-sans-paille — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have one major concern considering the API, but that's trivial to fix.

**Comment by jatinchowdhury18** [comment]
Ah, good call, I hadn't realized that my "phase-only" implementation was not part of the standard library. I've removed the phase-only implementation, and tweaked the default arguments of the other implementation to match the standard API.

**Comment by serge-sans-paille** [comment]
Thanks for this patch o/

**Pull request merged**